modules/extension/mbstyle/src/main/java/org/geotools/mbstyle/expression/MBExpression.java

```diff
@@ -151,24 +162,25 @@ public static MBExpression create(JSONArray json) {
             } else if (zoom.contains(name)) {
                 return new MBZoom(json);
             } else {
-                throw new MBFormatException("Expression \"" + name + "\" invalid.");
+                throw new MBFormatException("Expression \"" + name + "\" not supported.");
```

Comment:
This change isn't much of an improvement, as either could be true.
Given that GT doesn't necessarily support the full set of expressions, this distinction is relevant, because the invalid and not supported have different fixes - and invalid expression is user error (most likely a misspelling), and should be simple to fix, whereas an expression that is not supported means they need to upgrade (or wait for) a version of geotools that supports the expression.

Unfortunately I don't think there's much we can do about this - even if we maintained a list of expressions that GT didn't yet support, it would inevitably become outdated. Perhaps a more descriptive error message? Something like:
```
Expression \"" + name + "\" invalid. It may be misspelled or not supported by this implementation"
```

Reply:
Thanks, making the change you suggested.

The only positive improvement we could make here is to report the list of `data expressions` we do support; but that is likely to get long over time (and less useful as a result).